specification/search/data-plane/Microsoft.Azure.Search.Service/track1/preview/2019-05-06-preview/searchservice.json

```diff
@@ -4886,6 +4922,126 @@
         "key"
       ]
     },
+    "KnowledgeStore" :
+    {
+      "properties" : {
+        "storageConnectionString": {
+          "type": "string",
+          "description": "The connection string to the storage account projections will be stored in."
+        },
+        "projections": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KnowledgeStoreProjection",
+            "x-nullable": false
+          },
+          "description": "A list of additional projections to perform during indexing."
+        }
+      },
+      "required": [
+        "storageConnectionString",
+        "projections"
+      ],
+      "externalDocs": {
+        "url": "https://docs.microsoft.com/azure/search/knowledge-store-projection-overview"
+      }
+    },
+    "KnowledgeStoreProjection" :
+    {
+      "properties" : {
+        "tables": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KnowledgeStoreTableProjectionSelector",
+            "x-nullable": false
+          }
+        },
+        "objects": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KnowledgeStoreObjectProjectionSelector",
+            "x-nullable": false
+          }
+        },
+        "files": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KnowledgeStoreFileProjectionSelector",
+            "x-nullable": false
+          }
+        }
+      }
+    },
+    "KnowledgeStoreProjectionSelector" : {
+      "properties" : {
+        "referenceKeyName" : {
+          "type" : "string"
+        },
+        "generatedKeyName" : {
+          "type" : "string"
+        },
+        "source" : {
+          "type" : "string"
+        },
+        "sourceContext" : {
+          "type" : "string"
+        },
+        "inputs" : {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/InputFieldMappingEntry"
+          }
+        }
+      }
+    },
+    "KnowledgeStoreBlobProjectionSelector" : {
+      "properties" : {
+        "storageContainer" : {
+          "type" : "string"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/KnowledgeStoreProjectionSelector"
+        }
+      ],
+      "required": [
+        "storageContainer"
+      ]
```

Comment:
could we add a description here as well?

Reply:
do you want a comment to the effect that its an intermediate abstract class, and not to be instantiated directly?

---
In reply to: [349261085](https://github.com/Azure/azure-rest-api-specs/pull/7812#discussion_r349261085) [](ancestors = 349261085)

Reply:
Yeah, similar to how we do for the "Skill" model...

Reply:
I don’t think we should be talking about abstract classes in these descriptions. The swagger spec is not only used to generate .NET documentation, but documentation for other languages as well where the notion of an abstract class might not exist. Also, in the future, we will generate our REST API documentation from the spec. In that context, talking about abstract classes definitely does not make sense.

Reply:
Soo... we have a bunch of references to "Abstract base class..." in this spec (I assume Skills followed the existing pattern). Given that this is the track1 spec, which is focused primarily (or rather *only*) for .NET SDK, I think it's reasonable to follow that pattern.

When we are ready to start working on the "track2" spec, we should consider removing the word abstract

Reply:
The track 2 spec already exists, and it's being used for the Java SDK.

Reply:
@brjohnstmsft Let @jadrefke know what to do here - this is I think the only place left without a description

Reply:
@brjohnstmsft - This is the track1 preview spec.
I'd like Austin to continue using the same terms ("Abstract")...

The track2 spec that exists isn't the preview spec. It's the GA spec. Someone will need to make sure that we remove references to terms like Abstract.

When we have a track2 "preview" spec, we'll be careful to rephrase these new terms accordingly. I don't think we should block this PR for that

Reply:
@arv100kri Agreed, no need to block the PR on this.